ESACCI/read-biomass-ESACCI.py

```diff
@@ -0,0 +1,255 @@
+import numpy as np
+from netCDF4 import Dataset
+import pylab as plt
+import os
+import sys
+import time
+import datetime
+from mpl_toolkits.basemap import Basemap
+import math
+import glob
+from urllib.request import urlretrieve
+from subroutines import *
+from PIL import Image
+from pylab import *
+from osgeo import gdal, ogr, osr
+
+# set up Data directory 
+DataDir = "/Users/mingquan/newDATA"
+
+# set up initial and final years of data
+start_yr = 2017
+end_yr   = 2017
+
+remote_source = "https://climate.esa.int/en/odp/#/project"
+gist_source   = "https://github.com/mmu2019/Datasets/blob/master/read-biomass-ESACCI.py"
+local_source  = 'ESACCI-BIOMASS-L4-AGB-MERGED-100m-2017-fv1.0.nc'
+stamp1        = '2020-09-18'
+stamp2        = '2019-03-14'
+
+sourceID = "ESACCI"
+instit1  = "United space in Europe"
+instit2  = "The European Space Agency's (ESA's) Climate Change Initiative (CCI) programme, the Biomass CCI team"
+
+datestr = str(datetime.datetime.now())
+TmpStr  = datestr.split(' ')
+stamp3  = TmpStr[0]
+
+long_name = 'global forest above ground biomass'
+period = "2016-12-31-2017-12-30"
+origtr = "climatology"
+origsr = "100 m"
+origut = "Mg/ha biomass"
+finltr = "2016-12-31-2017-12-30"
+finlsr = "0.5 degree"
+finlut = "kg/m2 carbon"
+
+# Create temporal dimension
+nyears    = end_yr - start_yr + 1
+nmonth    = 12
+ndays  = np.asarray([31,28,31,30,31,30,31,31,30,31,30,31],dtype=float)
+smonth = np.asarray(['01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11', '12'])
+month_bnd = np.asarray([0,365],dtype=float)
+tbnd0  = np.asarray([((np.arange(nyears)*365)[:,np.newaxis]+month_bnd[:-1]).flatten(),
+                    ((np.arange(nyears)*365)[:,np.newaxis]+month_bnd[+1:]).flatten()]).T
+tbnd0 += (start_yr-1850)*365
+tbnd0.shape
+
+tbnd   = np.ma.masked_array(np.random.rand(1,2))
+
+tbnd[0,0] = tbnd0[0,0]
+tbnd[0,1] = tbnd0[nyears-1,1]
+
+t     = tbnd.mean(axis=1)
+t.shape
+
+# Create new spatial dimension
+res    = 0.5
+latbnd = np.asarray([np.arange(- 90    , 90     ,res),
+                     np.arange(- 90+res, 90+0.01,res)]).T
+lonbnd = np.asarray([np.arange(-180    ,180     ,res),
+                     np.arange(-180+res,180+0.01,res)]).T
+lat    = latbnd.mean(axis=1)
+lon    = lonbnd.mean(axis=1)
+
+# Create some fake data
+data   = np.ma.masked_array(np.random.rand(t.size,lat.size,lon.size))
+area   = np.ma.masked_array(np.random.rand(lat.size,lon.size))
```

Comment:
Here we want to initialize these arrays with zeros, not random numbers. 
`np.random.rand(t.size,lat.size,lon.size)` becomes `np.zeros((t.size,lat.size,lon.size))`. Note that the zeros functions wants the sizes inside its own tuple.